### DISCUSSION TITLE
Trigger TeslaMate to wake up?

### DISCUSSION THREAD

**Discussion by tobiasehlert** [teslamate-org/teslamate#1433] [Q&A]
I am wondering, if there is some way to trigger TeslaMate to wake up and resume logging with a HTTP POST request or so.
If I recall, there was something available like one year ago.. someone remembering?

_Why?_
If I wake up the car by an API call (not through TeslaMate) then TeslaMate continues sleeping for minutes to go. Therefore I want to trigger TeslaMate to resume logging when I want it to.

**Comment by tobiasehlert** [comment]
Maybe @adriankumpf can recall if that is something that is not left anymore or if there is a possible solution?

**Reply by eliluong** [reply]
I see snippets of code with a resume logging function, but I'm not familiar with the code to determine if it's something that actually is still working or how it is used. I think it's definitely doing something to signify "Driving / Start" based on the logs.

https://github.com/adriankumpf/teslamate/search?q=repo%3Aadriankumpf%2Fteslamate+resume
https://github.com/adriankumpf/teslamate/issues/102
https://github.com/adriankumpf/teslamate/issues/100

**Reply by adriankumpf** [reply]
For such cases there is `PUT /api/car/1/logging/resume`. It will instruct TeslaMate to resume logging and increase the polling interval for a short period.

**Reply by tobiasehlert** [reply]
> For such cases there is `PUT /api/car/1/logging/resume`. It will instruct TeslaMate to resume logging and increase the polling interval for a short period.

Alright @adriankumpf 
So there is an API endpoint `/api/car/...` which means that I'll have to write some code in TeslaMateApi to pass those two existing PUT requests to TeslaMate instead. Thanks for that info :)

**Comment by steve28** [comment]
I came here looking for this - tried teslamate some time ago and remember this - now I don't see it anywhere.  Also you used to be able to configure sleep durations and the like

**Reply by adriankumpf** [reply]
If the Streaming API feature is enabled (which is the default), you don't need to configure anything.

**Reply by steve28** [reply]
So when teslamate shows "Falling asleep for X min", if I started a drive then, it would catch all of it?

**Reply by adriankumpf** [reply]
Exactly!

**Reply by steve28** [reply]
sorry for being slow... then what is happening during the "falling asleep ..." phase?  It sounds like with the streaming API there is no need to pause anything...?

**Comment by ToniA** [comment]
Yesterday I added this into crontab to assist logging of scheduled charges (starts at midnight):

`58 23 * * *  /usr/bin/curl -X PUT http://localhost:4000/api/car/1/logging/resume`

Seems to do the trick, although I was not charging last night:
```
teslamate_1  | 2021-03-14 21:47:35.713 car_id=1 [info] Start / :asleep
teslamate_1  | 2021-03-14 21:47:35.816 car_id=1 [info] Disconnecting ...
teslamate_1  | 2021-03-14 23:58:03.046 [info] PUT /api/car/1/logging/resume
teslamate_1  | 2021-03-14 23:58:03.111 car_id=1 [info] Expecting imminent wakeup. Increasing polling frequency ...
teslamate_1  | 2021-03-14 23:58:03.112 [info] Sent 204 in 49ms
teslamate_1  | 2021-03-15 07:49:50.722 car_id=1 [info] Start / :online
teslamate_1  | 2021-03-15 07:49:50.985 car_id=1 [info] Connecting ...
```

@adriankumpf how long does Teslamate stay active before trying to put the car asleep if this is called?

**Reply by adriankumpf** [reply]
Three minutes by default. You'll find `[info] Suspending logging` in the logs.

**Reply by ToniA** [reply]
I think it would be nice to have an enhancement where you could give the intended online time as a parameter.

The use case is the slow MCU1 / instrument cluster startup from deep sleep. I was thinking of adding some automatic wakeups now that my car is finally sleeping:
* call the Tesla vehicle wake-up api
* instruct Teslamate to do logging for a given period of time, as a desired side-effect this keeps the car from falling asleep

I already tested that my car will stay awake with a shell script like this:
`while true; do sleep 60; curl -X PUT http://localhost:4000/api/car/1/logging/resume; done`
